datafusion/common/src/scalar.rs

```diff
@@ -1802,52 +1658,12 @@ impl ScalarValue {
     /// assert_eq!(result, &expected);
     /// ```
     pub fn new_list(values: &[ScalarValue], data_type: &DataType) -> ArrayRef {
-        Arc::new(match data_type {
```

Comment:
I am not sure why this function was implemented like this, but it is much more concisely (and completely) implemented in terms of arrays and offsets.

Reply:
Nice clean up.